src/LaunchDarkly.EventSource/EventSourceService.cs

```diff
@@ -208,7 +208,7 @@ private async Task<T> DoTaskWithReadTimeout<T>(Func<Task<T>> task, bool needManu
             }
 
             var timeoutCancellation = new CancellationTokenSource();
-            var readTimeoutTask = Task.Delay(_configuration.ReadTimeout, timeoutCancellation.Token);
+            var readTimeoutTask = Task.Run(() => Task.Delay(_configuration.ReadTimeout, timeoutCancellation.Token));
```

Comment:
I'm not 100% sure but it does seem as if using `Task.Run`, rather than just creating a task with a method like `Delay`, causes scheduling to be more predictable.

Reply:
That turned out to be not quite the issue, but I'll explain in the next PR where (I think) I greatly improved how timeouts work.